### PR TITLE
[GEOS-9207] REST exception handler and controllers do not always set the response content type

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/extension/geofence-server/src/main/java/org/geoserver/geofence/server/rest/RulesRestController.java [hunk 1]
```diff
@@ -206,7 +208,9 @@ public JaxbRuleList count(
             adminService.setDetails(id, rule.getLayerDetails().toLayerDetails(null));
         }
 
-        return new ResponseEntity<Long>(id, HttpStatus.CREATED);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.TEXT_PLAIN);
```
Comment (sikeoka) [inline src/extension/geofence-server/src/main/java/org/geoserver/geofence/server/rest/RulesRestController.java]: These headers aren't used.

File: src/rest/src/main/java/org/geoserver/rest/RestInterceptor.java [hunk 2]
```diff
@@ -23,4 +24,14 @@ public boolean preHandle(
...
+
+    @Override
+    public void postHandle(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            Object handler,
+            ModelAndView modelAndView)
+            throws Exception {
+        super.postHandle(request, response, handler, modelAndView);
+    }
```
Comment (sikeoka) [inline src/rest/src/main/java/org/geoserver/rest/RestInterceptor.java]: This doesn't seem to be doing anything useful?

File: src/restconfig/src/test/java/org/geoserver/rest/catalog/StyleControllerTest.java [hunk 3]
```diff
@@ -416,7 +424,7 @@ public void testPostToWorkspace() throws Exception {
         String xml = "<style>" + "<name>foo</name>" + "<filename>foo.sld</filename>" + "</style>";
         MockHttpServletResponse response =
                 postAsServletResponse(RestBaseController.ROOT_PATH + "/workspaces/gs/styles", xml);
-        assertEquals(201, response.getStatus());
```
Comment (sikeoka) [inline src/restconfig/src/test/java/org/geoserver/rest/catalog/StyleControllerTest.java]: Is removing the assert here necessary?

File: src/extension/geofence-server/src/main/java/org/geoserver/geofence/server/rest/RulesRestController.java (hunk 1 above)
Comment (aaime) [reply src/extension/geofence-server/src/main/java/org/geoserver/geofence/server/rest/RulesRestController.java]: Right, the change in "produces" does the trick, not this bit, I'll remove.

File: src/rest/src/main/java/org/geoserver/rest/RestInterceptor.java (hunk 2 above)
Comment (aaime) [reply src/rest/src/main/java/org/geoserver/rest/RestInterceptor.java]: True, leftover from a failed approach.

File: src/restconfig/src/test/java/org/geoserver/rest/catalog/StyleControllerTest.java (hunk 3 above)
Comment (aaime) [reply src/restconfig/src/test/java/org/geoserver/rest/catalog/StyleControllerTest.java]: No, copy/paste error while I was adding the assertions around.

**Comment by aaime** [comment]
Feedback from @sikeoka review applied.

**Pull request merged**